src/extensions/layout-selector/layout-selector-results.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo } from '@wordpress/element';
+import { useAsyncList } from '@wordpress/compose';
+import { Button, Spinner } from '@wordpress/components';
+import { BlockPreview } from '@wordpress/block-editor';
+import { applyFilters } from '@wordpress/hooks';
+
+/**
+ * Render the layout previews into columns.
+ */
+export const LayoutSelectorResults = ( { layouts, category, onInsert } ) => {
+	const filteredLayouts = useMemo(
+		() => layouts
+			.filter( ( layout ) => layout.category === category )
+			.map( ( layout ) => ( { ...layout,
+				/**
+				 * Filters the list of blocks within the layout preview.
+				 *
+				 * @param {Array} blocks The block objects of the layout.
+				 */
+				blocks: applyFilters( 'coblocks.layoutPreviewBlocks', layout.blocks ),
+			} ) ),
+		[ layouts, category ]
+	);
+
+	// Needed to render our list of previews asynchronously for better performance.
+	const currentShowLayouts = useAsyncList( filteredLayouts );
```

Comment:
Oh that's pretty cool. Nice find

Reply:
Yeah, it was buried in the patterns api code. Works well :)